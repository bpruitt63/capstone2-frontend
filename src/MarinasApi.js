import axios from 'axios';
//import {marinas} from './ApiKeys';

const BASE_URL = 'https://api.marinas.com/v1/';
const API_KEY = process.env.REACT_APP_MARINAS_API_KEY;

class MarinasApi {

    /** Retrieves array of all points of interest within a certain distance
     * of a specific latitude/longitude.
     */
    static async getPoints(longLat) {
        try {
            const res = await axios.get(
                `${BASE_URL}points/search?location[lon]=${longLat[0]}&location[lat]=${longLat[1]}&location[radius]=20000&access_key=${API_KEY}`
            );
            return res;
        } catch(err) {
            console.error("marinas api Error:", err.response);
            let message;
            !err.response ? message = "Server error, please try again later"
                : message = err.response.data.error.message;
            throw Array.isArray(message) ? message : [message];
        };
    };

    /** Retrieves info on a specific point of interest */
    static async getOnePoint(id) {
        try {
            const res = await axios.get(
                `${BASE_URL}points/${id}`
            );
            return res.data;
        } catch(err) {
            console.error("marinas api Error:", err.response);
            let message;
            !err.response ? message = "Server error, please try again later"
                : message = err.response.data.error.message;
            throw Array.isArray(message) ? message : [message];
        };
    };
};

export default MarinasApi;