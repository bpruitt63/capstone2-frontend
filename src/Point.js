import React, {useState} from 'react';
import PointDetails from './PointDetails';

function Point({point}) {

    const [showDetails, setShowDetails] = useState(false);

    const toggle = () => {
        setShowDetails(!showDetails);
    };

    return (
        <div>
            <h4>{point.name}</h4>
            <p>Location type:  {point.kind[0].toUpperCase() + point.kind.substring(1)}</p>
            <button onClick={toggle}>
                {showDetails ? 'Hide Details' : 'Show Details'}
            </button>
            {showDetails && <PointDetails point={point} />}
        </div>
    );
};

export default Point;