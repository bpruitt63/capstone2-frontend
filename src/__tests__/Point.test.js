import React from 'react';
import { render, screen, fireEvent, waitForElementToBeRemoved } from '@testing-library/react';
import Point from '../Point';
import MarinasApi from '../MarinasApi';
import { BrowserRouter } from 'react-router-dom';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
    Map: () => ({}),
}));

jest.mock('../MarinasApi');

const searchedPoint = {id: 'test1', 
                kind: 'marina',
                name: 'Test Location',
                review_count: 0,
                web_url: 'https://google.com',
                location: {lat: 40, lon: 10},
                fuel: {has_diesel: false,
                        has_propane: false,
                        has_gas: true},
                images: {data: []}};

const savedPoint = {locationId: 'test2',
                    locationPosition: 0, 
                    locationName: 'Test Location 2'};

test('renders', () => {
    render(<BrowserRouter><Point point={searchedPoint}/></BrowserRouter>);
});

test('matches snapshot', () => {
    const {asFragment} = render(<BrowserRouter><Point point={searchedPoint}/></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
});

test('renders point of interest', () => {
    render(<BrowserRouter><Point point={searchedPoint} searchHere={{}}/></BrowserRouter>);
    expect(screen.getByText(/Marina/)).toBeInTheDocument;
    expect(screen.getByText(/Center Search Here/)).toBeInTheDocument;
});

test('renders recent trip point and shows details', async () => {
    MarinasApi.getOnePoint.mockResolvedValue(searchedPoint);
    render(<BrowserRouter><Point point={savedPoint} isSaved={true}/></BrowserRouter>);
    expect(screen.queryByText(/Marina/)).not.toBeInTheDocument;
    expect(screen.getByText(/Test Location 2/)).toBeInTheDocument;
    fireEvent.click(screen.getByText(/Show Details/));
    await waitForElementToBeRemoved(() => screen.getByText(/Show Details/));
    expect(screen.getByText(/Fuel available/)).toBeInTheDocument;
    expect(screen.queryByText(/Your Rating:/)).not.toBeInTheDocument;
});

test('renders user trip point and shows details', async () => {
    MarinasApi.getOnePoint.mockResolvedValue(searchedPoint);
    render(<BrowserRouter><Point point={savedPoint} isSaved={true} username={'test'}/></BrowserRouter>);
    expect(screen.queryByText(/Marina/)).not.toBeInTheDocument;
    expect(screen.getByText(/Test Location 2/)).toBeInTheDocument;
    fireEvent.click(screen.getByText(/Show Details/));
    await waitForElementToBeRemoved(() => screen.getByText(/Show Details/));
    expect(screen.getByText(/Fuel available/)).toBeInTheDocument;
    expect(screen.getByText(/Your Rating:/)).toBeInTheDocument;
});