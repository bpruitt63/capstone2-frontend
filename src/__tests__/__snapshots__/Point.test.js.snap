// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="searchedPoint"
  >
    <h4
      class="pointTitle"
    >
      Test Location
    </h4>
    <p>
      <span
        class="detailTitle"
      >
        Location type: 
      </span>
      Marina
    </p>
    <div
      class="collapse"
    />
    <button
      class="pointButton btn btn-secondary"
      type="button"
    >
      Show Details
    </button>
  </div>
</DocumentFragment>
`;
